### PR TITLE
test(ctb): bridge return to sender on fail

### PR DESCRIPTION
**Pull request by smartcontracts**
**Description**
Adds a test that bridge contracts return to sender on failed deposits or
withdrawals.

### CODE REVIEW

File: packages/contracts-bedrock/contracts/test/L1StandardBridge.t.sol [hunk 1]
```diff
@@ -391,4 +391,48 @@ contract L1StandardBridge_Test is Bridge_Initializer {
...
+        vm.store(address(L1Bridge), bytes32(slot), bytes32(uint256(100)));
+        assertEq(L1Bridge.deposits(address(BadL1Token), address(L2Token)), 100);
+
+        vm.expectEmit(true, true, true, true);
+
+        emit ERC20BridgeInitiated(
+            address(BadL1Token),
+            address(L2Token),
+            alice,
+            alice,
```
Comment (maurelian) [inline packages/contracts-bedrock/contracts/test/L1StandardBridge.t.sol:415]: This misses the possibility (which we've gotten wrong previously), that the tokens are returned to the wrong sender on the other domain. I'd suggest this change and the one below:
```suggestion
            bob,
            alice,
```

File: packages/contracts-bedrock/contracts/test/L1StandardBridge.t.sol [hunk 2]
```diff
@@ -391,4 +391,48 @@ contract L1StandardBridge_Test is Bridge_Initializer {
...
+            address(L1Bridge.messenger()),
+            abi.encodeWithSelector(CrossDomainMessenger.xDomainMessageSender.selector),
+            abi.encode(address(L1Bridge.otherBridge()))
+        );
+        vm.prank(address(L1Bridge.messenger()));
+        L1Bridge.finalizeBridgeERC20(
+            address(BadL1Token),
+            address(L2Token),
+            alice,
+            alice,
```
Comment (maurelian) [inline packages/contracts-bedrock/contracts/test/L1StandardBridge.t.sol:430]: I think this should be 
```suggestion
            alice,
            bob,
```

File: packages/contracts-bedrock/contracts/test/L1StandardBridge.t.sol (hunk 1 above)
Comment (smartcontracts) [reply packages/contracts-bedrock/contracts/test/L1StandardBridge.t.sol]: Done

Comment (maurelian) [reply packages/contracts-bedrock/contracts/test/L1StandardBridge.t.sol]: jc that's fast

**Review by maurelian — APPROVED** [review]

**Review by tynes — APPROVED** [review]

**Pull request merged**